### PR TITLE
feat: replacing the map types  (#1365)

### PR DESCRIPTION
**Pull request by dhruvmalik007**
- replacing the HashMap and HashSet from the std::collections to the alloy-primitives along with setting up the cargo file.

- setting the dependencies in the cargo files

- [ ]Bug fixes and new features should include tests: TBD.

## Motivation

to close the PR #1365 . 

## Solution

migrating the types as defined in the given issue.

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types-eth/src/lib.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ extern crate alloc;
 pub mod collections {
```
Comment (mattsse) [inline crates/rpc-types-eth/src/lib.rs:15]: I believe this is now obsolete?

File: crates/rpc-types/Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,7 @@ alloy-rpc-types-mev = { workspace = true, optional = true }
 alloy-rpc-types-trace = { workspace = true, optional = true }
 alloy-rpc-types-txpool = { workspace = true, optional = true }
 serde = { workspace = true, features = ["derive", "std"] }
+alloy-primitives = { version = "0.8.5", features = ["map"] }
```
Comment (mattsse) [inline crates/rpc-types/Cargo.toml:33]: this should use workspace

File: crates/rpc-types/src/rpc.rs [hunk 3]
```diff
@@ -28,13 +27,16 @@ mod tests {
     #[test]
     fn test_parse_module_versions_roundtrip() {
         let s = r#"{"txpool":"1.0","trace":"1.0","eth":"1.0","web3":"1.0","net":"1.0"}"#;
-        let module_map = HashMap::from([
+        let mut module_map: HashMap<String, String> = HashMap::default(); // Change to FxHashMap
+
+         module_map.extend([
```
Comment (mattsse) [inline crates/rpc-types/src/rpc.rs:32]: this should use `from_iter`

**Review by mattsse — CHANGES_REQUESTED** [review]
needs closer look by @DaniPopes

File: crates/rpc-types-eth/src/lib.rs (hunk 1 above)
Comment (dhruvmalik007) [reply crates/rpc-types-eth/src/lib.rs:15]: yes Its removed as redundant.

**Pull request closed without merging**